app/apptesting/test_suite.go

```diff
@@ -472,8 +472,14 @@ func (s *KeeperTestHelper) RunMsg(msg sdk.Msg) (*sdk.Result, error) {
 	// cursed that we have to copy this internal logic from SDK
 	router := s.App.GetBaseApp().MsgServiceRouter()
 	if handler := router.Handler(msg); handler != nil {
+		// Use a cache context to only write if the handler is successful
+		cacheCtx, writeCache := s.Ctx.CacheContext()
```

Comment:
This is needed to allow failing txs. Before, we were only using the integration test utils on successes.